HARK/distributions/continuous.py

```diff
@@ -55,14 +55,43 @@ class Normal(ContinuousFrozenDistribution):
         Seed for random number generator.
     """
 
-    def __init__(self, mu=0.0, sigma=1.0, seed=0):
+    def __new__(
+        cls,
+        mu: Union[float, np.ndarray] = 0.0,
+        sigma: Union[float, np.ndarray] = 1.0,
+        seed: Optional[int] = None,
+    ):
+        """
+        Create a new Normal distribution. If sigma is zero, return a
+        DiscreteDistribution with a single atom at mu.
+        Parameters
+        ----------
+        mu : Union[float, np.ndarray], optional
+            Mean of normal distribution, by default 0.0
+        sigma : Union[float, np.ndarray], optional
+            Standard deviation of normal distribution, by default 1.0
+        seed : Optional[int], optional
+            Seed for random number generator, by default None
+        Returns
+        -------
+        Normal or DiscreteDistribution
+            Normal distribution or DiscreteDistribution with a single atom.
+        """
```

Comment:
Nice pattern - handling the degenerate case (sigma=0) by returning a DiscreteDistribution instead of a continuous one.